superset-frontend/src/explore/AdhocFilter.js

```diff
@@ -48,22 +48,13 @@ const OPERATORS_TO_SQL = {
 };
 
 function translateToSql(adhocMetric, { useSimple } = {}) {
-  if (
-    (adhocMetric.expressionType === EXPRESSION_TYPES.SIMPLE &&
-      adhocMetric.comparator &&
-      adhocMetric.operator) ||
-    useSimple
-  ) {
-    const isMulti = MULTI_OPERATORS.has(adhocMetric.operator);
-    const { subject } = adhocMetric;
+  if (adhocMetric.expressionType === EXPRESSION_TYPES.SIMPLE || useSimple) {
+    const { subject, comparator } = adhocMetric;
```

Comment:
`getSimpleSQLExpression is now able to handle invalid formatting requests, hence this strict checking isn't necessary anymore (no translation was performed if operator wasn't specified).